### ISSUE TITLE
Switch to the rust-vmm vm-virtio crate

### ISSUE THREAD

**Issue by sameo** [cloud-hypervisor/cloud-hypervisor#1413]
In order to factorize the core virtio code (queue management, virtio device trait), we should eventually depend on the upstream rust-vmm vm-virtio crate.

Proposal to reach that goal:

1. #1411 
1. Sync the `cloud-hypervisor/vm-virtio` crate's `ch` branch with the `cloud-hypervisor/cloud-hypervisor/vm-virtio` crate
1. Switch the `cloud-hypervisor/cloud-hypervisor` code to depend on the `cloud-hypervisor/vm-virtio` `ch` branch.
1. Submit the `ch` branch upstream
1. Fingers crossed for getting our contributions merged...
1. Switch to the rust-vmm vm-virtio upstream crate

Depends on #1411

**Comment by rbradford** [comment]
Superseded by #2055 which captures the same issue and the necessary collateral change to switch to vhost-user-backend
